sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/TracerProviderConfiguration.java

```diff
@@ -115,6 +119,14 @@ static SpanLimits configureSpanLimits(ConfigProperties config) {
 
   // Visible for testing
   static Sampler configureSampler(String sampler, ConfigProperties config) {
+    Map<String, Sampler> spiSamplers =
+        StreamSupport.stream(
+                ServiceLoader.load(ConfigurableSamplerProvider.class).spliterator(), false)
+            .collect(
+                Collectors.toMap(
+                    ConfigurableSamplerProvider::getName,
+                    provider -> provider.createSampler(config)));
+
```

Comment:
could move this into the `default` branch and only instantiate the requested sampler

Reply:
This is our current pattern will stick with it for this PR but seems like a good improvement for all of em :)